docs/tutorials/data_provider.md

```diff
@@ -269,31 +270,35 @@ use tinystr::tinystr;
 
 pub struct CustomDecimalSymbolsProvider<P>(P);
 
-impl<P> AnyProvider for CustomDecimalSymbolsProvider<P>
+impl<P, M> DataProvider<M> for CustomDecimalSymbolsProvider<P>
 where
-    P: AnyProvider
+    P: DataProvider<M>,
+    M: KeyedDataMarker,
 {
-    fn load_any(&self, key: DataKey, req: DataRequest) -> Result<AnyResponse, DataError> {
-        let mut any_res = self.0.load_any(key, req)?;
-        if key == DecimalSymbolsV1Marker::KEY && req.locale.region() == Some(region!("CH")) {
-            let mut res: DataResponse<DecimalSymbolsV1Marker> = any_res.downcast()?;
-            if let Some(payload) = &mut res.payload.as_mut() {
-                payload.with_mut(|data| {
-                    // Change the grouping separator for all Swiss locales to '🐮'
-                    data.grouping_separator = Cow::Borrowed("🐮");
-                });
+    #[inline]
+    fn load(&self, req: DataRequest) -> Result<DataResponse<M>, DataError> {
+        let mut res = self.0.load(req)?;
+        if let Some(mut generic_payload) = res.payload.as_mut() {
+            // Cast from `DataPayload<M>` to `DataPayload<DecimalSymbolsV1Marker>`
+            let mut any_payload = generic_payload as &mut dyn Any;
+            if let Some(mut decimal_payload) = any_payload.downcast_mut::<DataPayload<DecimalSymbolsV1Marker>>() {
+                if req.locale.region() == Some(region!("CH")) {
```

Comment:
put the `CH` check at the top level?

Reply:
I'd rather keep the region nearer the payload override for readability. I don't have reason to believe that there's a performance difference between checking the TypeId of the payload versus the region; maybe there's a small difference but this code needs to exist either way.